avro2py/codegen.py

```diff
@@ -408,6 +408,25 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
             return rewriter.visit(node)
 
 
+def resolve_schema_reference(schema: Record, schemas : List[Record]):
+    """Substitute the schema reference with the definition.""" 
+    if not schemas:
+        return
+
+    for field in schema.resolved_schema['fields']:
+        if isinstance(field['type'], str) and field['type'].find('.') >= 0:
+            ref_schema_name = field['type'].split('.')[-1]
+            is_resolved = False
+            for s in schemas:
+                if s.name == ref_schema_name:
+                    resolve_schema_reference(s, schemas)
```

Comment:
This is going to mean that we're resolving things multiple times over, is there no caching mechanism we can employ?

Reply:
Used a `set()` as a cache to store fully qualified name of resolved schemas.